builder/vsphere/clone/step_customize.go

```diff
@@ -92,37 +121,34 @@ type StepCustomize struct {
 func (c *CustomizeConfig) Prepare() []error {
 	var errs []error
 
-	if c.LinuxOptions == nil && c.WindowsSysPrepFile == "" {
-		errs = append(errs, fmt.Errorf("customize is empty"))
+	options_number := 0
+	if c.LinuxOptions != nil {
+		options_number = options_number + 1
 	}
-	if c.LinuxOptions != nil && c.WindowsSysPrepFile != "" {
-		errs = append(errs, fmt.Errorf("`linux_options` and `windows_sysprep_text` both set - one must not be included if the other is specified"))
+	if c.WindowsOptions != nil {
+		options_number = options_number + 1
+	}
+	if c.WindowsSysPrepFile != "" {
+		options_number = options_number + 1
 	}
 
-	if c.LinuxOptions != nil {
-		if c.LinuxOptions.Hostname == "" {
-			errs = append(errs, fmt.Errorf("linux options `host_name` is empty"))
-		}
-		if c.LinuxOptions.Domain == "" {
-			errs = append(errs, fmt.Errorf("linux options `domain` is empty"))
-		}
-
-		if c.LinuxOptions.HWClockUTC == config.TriUnset {
-			c.LinuxOptions.HWClockUTC = config.TriTrue
-		}
-		if c.LinuxOptions.Timezone == "" {
-			c.LinuxOptions.Timezone = "UTC"
-		}
+	if options_number > 1 {
+		errs = append(errs, fmt.Errorf("Only one of `linux_options`, `windows_options`, `windows_sysprep_file` can be set"))
+	} else if options_number == 0 {
+		errs = append(errs, fmt.Errorf("One of `linux_options`, `windows_options`, `windows_sysprep_file` must be set"))
 	}
 
-	if len(c.NetworkInterfaces) == 0 {
```

Comment:
❓I'm curious why this check for the number of interfaces was removed.

Reply:
I do wonder if it was just removed as part of the long if conditional block by accident.

Reply:
I'm thinking so. Should revert that change. I'd be happy to run some tests once I'm back online regularly in about two weeks.

Reply:
Thanks for adding it back @nywilken.

Reply:
It's a requirement to have one network_interface "defined" even if empty.

Reply:
Thanks for calling this out @tenthirtyam. I saw this requirement in the documentation. Which states "To use DHCP, declare an empty network_interface for each adapter being configured. This field is required". This overrides the network interface settings from the cloned vm correct?

Would it make more sense to not require an empty network configuration and just assume DHCP to start. If a users wishes to customize then the network_interface would not be empty and would work as it does now.

Reply:
I do think that would be a simpler approach, Wilken.